daemon/logger/ring.go

```diff
@@ -0,0 +1,219 @@
+package logger
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultRingMaxSize = 1000000 // 1MB
+)
+
+// RingLogger is a ring buffer that implements the Logger interface.
+// This is used when lossy logging is OK.
+type RingLogger struct {
+	buffer *messageRing
+	ctx    context.Context
```

Comment:
Yes I've heard this as well, but is really nice here because I don't have to worry about concurrent closes.
Probably have to add a separate mutex (or a `sync.Once`) just to handle close.

But :man_shrugging:
In the end, either work for me.